src/src/drivers/emulated/serial.c

```diff
@@ -0,0 +1,111 @@
+#include "serial.h"
+#include <omen/managers/dev/devices.h>
+
+// Define the file operations structure for the serial device
+struct file_operations serial_fops = {
+    .read = (size_t (*)(uint64_t, uint64_t, uint8_t *))serial_read,
+    .write = (size_t (*)(uint64_t, uint64_t, uint8_t *))serial_write,
+    .ioctl = (int (*)(uint64_t, uint64_t, void *))serial_ioctl
+};
+
+// Define and initialize the serial device structure
+static serial_port_t serial_device = {
+    .port = COM1_BASE // Use the base address for COM1
+};
+
+// Initialize the serial device and register it
+void init_serial_device(uint8_t com, uint16_t port_base) {
+    // Register the serial device with a major number (e.g., 1)
+    serial_device.port = port_base;
+
+    status_t status = register_char(com, "seriala", &serial_fops);
+    if (status != SUCCESS) {
+        kprintf("Failed to register serial device\n");
+//        return;
+    }
+
+    // Optionally create the device entry
+    char *device_name = device_create(&serial_device, 1, 0);
```

Comment:
this is where you create the device, but first you should make sure the device really is present by echoing a few characters